server/modules/surveyRdb/api/surveyRdbApi.js

```diff
@@ -34,6 +37,32 @@ export const init = (app) => {
     }
   })
 
+  app.post('/surveyRdb/:surveyId/:nodeDefUuidTable/chart', requireRecordListViewPermission, async (req, res, next) => {
+    try {
+      const { surveyId, cycle, query: queryParam, chart: chartSpec } = Request.getParams(req)
+
+      const query = A.parse(queryParam)
+      const chart = A.parse(chartSpec)
+
+      const rows = await SurveyRdbService.fetchViewData({ surveyId, cycle, query })  
+
+      const spec = {...chart, data: {
+        name: 'table',
+        values: rows
+      }}
+
+
+      // TODO extract to Chart Service to encapsulate Vega inside
+      let vegaspec = lite.compile(spec).spec
+      const view = new vega.View(vega.parse(vegaspec), { renderer: 'none' })
+      const svg = await view.toSVG()
+
+      res.json({ svg })
```

Comment:
ah ok, I see it, I would move both services and api to another namespace, something like dataCharts, or "reporting"